### ISSUE TITLE
Provide method for deploying multiple flows

### ISSUE THREAD

**Issue by gabcoyne** [PrefectHQ/prefect#7209]
### First check

- [x] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

Currently the supported methods for deployment support deploying a single flow at one time. In order to incorporate deployment in any automated release process some scripting is required.

### Describe the proposed behavior

Provide a consistent format, probably YAML to capture deployment information about a number of flows, example:
```
flows:
    name: cool-flow
    infrastructure-block: k8s/custom-k8s
    storage-block: s3/cool-bucket
    schedule: * * * * * 
```

### Example Use

Using a separate CLI arg we can build and deploy a number of flows in a single file system i.e. `prefect deployment bulk -f deployments.yaml`

**Comment by bennnym** [comment]
+1 for this - as we have migrated from prefect v1 we have experienced our CI build double in time. 

We use a similar pattern https://github.com/bennnym/prefect-ci-pattern.

Going to look at multi threading to decrease time but that comes at an engineering cost and complexity too, would be great for larger orgs to have that abstracted away.

**Comment by EmilRex** [comment]
I believe this is being addressed in #9168

**Comment by discdiver** [comment]
Indeed, https://github.com/PrefectHQ/prefect/issues/9168 allows multiple deployments to be built from a single project.
